OHIF/Viewers
# fix(viewports): The display of linked viewports during drag and drop has a race

### Context

If multiple react state events occur near each other, react can decide to combine this, or can serialize them, or any combination thereof.  As well, any number of renderings or none can occur in between these.  This combination can lead to viewports
being in the midst of being created or destroyed when being rendered, or a partial new rendering occur, causing exceptions.


### Changes & Results

There are two parts to the fix.  First, not reusing viewport Id's at all means that create/destroy events on single viewport id's don't occur.  Second, making the drag and drop and segmentation updates atomic (single operation done at once) means it occurs consistently and in the same order and doesn't invoke multiple renders.  This also, incidentally, makes it faster.

### Testing

To test, the easiest way is to open up the test:e2e:serve environment, and view the SEG containing series in basic test mode
Try it with various layouts and protocol stages with:
Seg in one viewport, primary images in one or two other viewports
Ensure that you can load the seg, and that after you can:
1. Add the SEG to a blank viewport
2. Navigate each image series separately
3. Change layouts still
4. Note that during changing layouts there are some issues with remembering the positioning of multiple viewports containing the same series.

### Checklist

#### PR



- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates



- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
